# Part -1 Python Basics (Variables)

1. Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [1]:
# Write your code here
print("Name: Muhammad Yousuf\nFather's Name: Fazal Hussain\nDate of Birth: 01-Jan-1999")


Name: Muhammad Yousuf
Father's Name: Fazal Hussain
Date of Birth: 01-Jan-1999


2. Write your small bio using variables and print it using print function

In [7]:
# Write your code here
name = 'Ali'
father = 'Ahmed'
age= 25
address= 'Bathisland'

print(name)
print('age',age)
print('address',address)


Ali
age 25
address Bathisland


3. Write a program in which use all the operators we can use in Python

In [12]:
# Write your code here

a=10
b=20
c=5

result = (a+b)*c
print(result)

150


4. Completes the following steps of small task:
    - Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
    - Mention Variable of Total Marks and assign 300 to it
    - Calculate Percentage

In [24]:
# Write your code here
# Pehle variables ko define karein
english = 75
islamiat = 85
math = 90
totalmarks = 300  # Total marks (100 for each subject)

obtained_marks = english + islamiat + math

print("Obtained Marks:", obtained_marks)

percentage = (obtained_marks / totalmarks) * 100
print("Percentage:", percentage)


Obtained Marks: 250
Percentage: 83.33333333333334


# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [30]:
#Type your code here

salary = int(input("Enter your current salary: "))
year = int(input("Enter your years of experience: "))

if year > 5:
    bonus = salary * 0.05
    print(f"Congratulations! Your bonus amount is: {bonus:.2f}")
else:
    print("Sorry, you are not eligible for a bonus.")


Enter your current salary:  50000
Enter your years of experience:  6


Congratulations! Your bonus amount is: 2500.00



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [44]:
#Type your code here

age = int(input("Enter your age:"))
if age > 18:
    print ('eligibble')
else: 
 print('Not eligible')


Enter your age: 19


eligibble



3) Write a program to check whether a number entered by user is even or odd.

In [52]:
#Type your code here

num = int(input('Enter Your number:'))

if num % 2==0:
    print("Even")
else: 
   print('odd')



Enter Your number: 11


odd


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [56]:
#Type your code here

num = int(input('Enter your number:'))

if num % 7 ==0:
    print('Divisibe')
else:
    print('Not Divisible')





Enter your number: 10


Not Divisible


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [62]:
#Type your code here

# Number user se input karein
number = int(input('Enter your number: '))

# Check whether the number is a multiple of 5
if number % 5 == 0:
    print('Hello')
else:
    print('Bye')



Enter your number:  10


Hello


7) Write a program to display the last digit of a number.


In [79]:
#Type your code here


number = int(input("Enter a number: "))
last_digit = number % 10
print(f"The last digit of the number {number} is {last_digit}.")



Enter a number:  2564


The last digit of the number 2564 is 4.


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [81]:
#Type your code here 

length = int (input('Enter a length'))
breadth = int (input('Enter a breadth'))

if length == breadth:
    print("The shape is square")

else: 
    print('The shape is rectangle')



Enter a length 11
Enter a breadth 11


The shape is square


10) Take two int values from user and print greatest among them.

In [93]:
# Pehle input user se lene
num1 = int(input("Enter a number1: "))
num2 = int(input("Enter a number2: "))

# Check which number is greater
if num1 > num2:
    print(f"{num1} is greater.")
else:
    print(f"{num2} is greater.")

                 

Enter a number1:  12
Enter a number2:  21


21 is greater.


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [95]:
#Type your code here

quantity = int(input("Enter the quantity: "))
unit_cost = 100
total_cost = quantity * unit_cost
if total_cost > 1000:
    total_cost *= 0.9  
print(f"The total cost for {quantity} units is: {total_cost}")





Enter the quantity:  100


The total cost for 100 units is: 9000.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [97]:
#Type your code here


marks = int(input("Enter the marks: "))


if marks < 25:
    print("Grade: F")
elif marks >= 25 and marks <= 45:
    print("Grade: E")
elif marks >= 45 and marks <= 50:
    print("Grade: D")
elif marks >= 50 and marks <= 60:
    print("Grade: C")
elif marks >= 60 and marks <= 80:
    print("Grade: B")
else:
    print("Grade: A")



Enter the marks:  25


Grade: E


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [99]:
#Type your code here


classes_held = int(input("Enter the number of classes held: "))
classes_attended = int(input("Enter the number of classes attended: "))

attendance_percentage = (classes_attended / classes_held) * 100

if attendance_percentage < 75:
    print(f"Attendance Percentage: {attendance_percentage:.2f}%")
    print("The student is not allowed to sit in the exam.")
else:
    print(f"Attendance Percentage: {attendance_percentage:.2f}%")
    print("The student is allowed to sit in the exam.")









Enter the number of classes held:  5
Enter the number of classes attended:  12


Attendance Percentage: 240.00%
The student is allowed to sit in the exam.


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [20]:
#Type your code here

16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [21]:
#Type your code here

17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [22]:
#Type your code here

6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [18]:
#Type your code here